src/main/java/io/cdap/plugin/google/shopping/content/source/ShoppingContentSplit.java

```diff
@@ -0,0 +1,68 @@
+package io.cdap.plugin.google.shopping.content.source;
+
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A input split containing products reading from Google Shopping Content API.
+ */
+public class ShoppingContentSplit extends InputSplit implements Writable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShoppingContentBatchSource.class);
+
+  private String serviceAccountPath;
+
+  private String merchantId;
+
+  private ShoppingContentClient client;
+
+  public ShoppingContentSplit(String serviceAccountPath, String merchantId) {
+    this.serviceAccountPath = serviceAccountPath;
+    this.merchantId = merchantId;
+  }
+
+  public ShoppingContentSplit() throws IOException, GeneralSecurityException {
+    // Default constructor here is needed for Hadoop deserialization.
+    this.client = new ShoppingContentClient(serviceAccountPath, merchantId);
+  }
+
+  public boolean hasNextPage() throws IOException {
```

Comment:
It's best for logic around reading data to go in the RecordReader and not the Split. A Split is only meant to hold information that is used by the RecordReader to read a subset of the input data. The RecordReader has an initialize() method that is used to create any resources required where it should create the ShoppingContentClient.

Reply:
done